edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.IoTHub/EdgeAgentConnection.cs

```diff
@@ -97,12 +104,16 @@ public class EdgeAgentConnection : IEdgeAgentConnection
             this.deploymentMetrics = Preconditions.CheckNotNull(deploymentMetrics, nameof(deploymentMetrics));
             this.initTask = this.ForceRefreshTwin();
             this.manifestTrustBundle = manifestTrustBundle;
+            this.TwinPullOnConnectThrottleTime = TimeSpan.FromSeconds(30);
         }
 
         public Option<TwinCollection> ReportedProperties => this.reportedProperties;
 
         public IModuleConnection ModuleConnection => this.moduleConnection;
 
+        // Used by tests to shorten delay time
+        public TimeSpan TwinPullOnConnectThrottleTime { get; set; }
```

Comment:
Create an internal ctor that accepts this value instead.

Reply:
fixed